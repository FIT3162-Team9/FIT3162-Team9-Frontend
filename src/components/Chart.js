import React, { PureComponent } from 'react';
import './../App.css'
import {
  AreaChart, Area, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer
} from 'recharts';
import moment from 'moment';
import Paper from '@material-ui/core/Paper';

function formatXAxis(tickItem) {
  return moment.unix(tickItem).format('YYYY-MM-DD')
}


export default ({ data }) => {
  
  return (
      <ResponsiveContainer height={300} width="95%">
        <AreaChart data={data} >
          <defs>
            <linearGradient id="colorUv" x1="0" y1="0" x2="0" y2="1">
              <stop offset="1%" stopColor="#FAD6A5" stopOpacity={0.8} />
              <stop offset="99%" stopColor="#FAD6A5" stopOpacity={0} />
            </linearGradient>
            <linearGradient id="colorPv" x1="0" y1="0" x2="0" y2="1">
              <stop offset="1%" stopColor="#ABD1F3" stopOpacity={0.8} />
              <stop offset="99%" stopColor="#ABD1F3" stopOpacity={0} />
            </linearGradient>
          </defs>
          <XAxis dataKey="timestamp" tickFormatter={formatXAxis} />
          <YAxis unit='Â°C' />
          <CartesianGrid strokeDasharray="3 3" />
          <Tooltip />
          <Area type="monotone" dataKey="min" stroke="#FAD6A5" fillOpacity={1} fill="url(#colorUv)" />
        </AreaChart>
      </ResponsiveContainer>
  )
}