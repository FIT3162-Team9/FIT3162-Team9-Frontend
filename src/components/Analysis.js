import React, {useState, useEffect} from "react";
import Chart from './AnalysisChart.js';
import { getTemperature } from './firebase';
import CircularProgress from '@material-ui/core/CircularProgress';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';

import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import RefreshIcon from '@material-ui/icons/Refresh';
import FormControl from '@material-ui/core/FormControl';

import DateRangePicker from '@wojtekmaj/react-daterange-picker';

function Analysis(props){
    
    return(
        <Grid xs={12} md={12} lg={12}>
            <Grid
                container
                xs={12}
                md={12}
                lg={12}
                justify="space-between"
                style={{paddingRight: 10, paddingLeft: 10, paddingTop: 10}}
            >
                <Typography>Bushfire Analysis</Typography>
                <div style={{justifyContent: 'center'}}>
                    {/* <FormControl size="small" style={{paddingRight: 10, paddingLeft: 10 }}>
                        <Select
                            labelId="station-id-label"
                            id="station-id"
                            value={props.method.stationId}
                            onChange={props.method.handleStationIdChange}
                        >
                            {props.method.validStationIds.map(id => <MenuItem value={id}>{id}</MenuItem>)}
                        </Select>
                    </FormControl> */}
                    <DateRangePicker
                        onChange={props.method.setDateRange}
                        value={props.method.dateRange}
                    />
                     <IconButton aria-label="refresh" onClick={props.method.refreshTemp}>
                        <RefreshIcon fontSize="small" />
                    </IconButton>
                </div>
            </Grid>
            <Grid item xs={12} md={12} lg={12}>
                {(props.method.tempData.length !== 0) ? (<Chart data={{data:props.method.tempData, state:props.method.state}}/>) : (<CircularProgress/>)}
            </Grid>
        </Grid>

    )
}

export default Analysis