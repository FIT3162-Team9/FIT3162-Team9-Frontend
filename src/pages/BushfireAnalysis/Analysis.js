import React, {useState, useEffect} from "react";
import Chart from './AnalysisChart.js';
import { getTemperature } from '../../helpers/TemperatureApi';
import CircularProgress from '@material-ui/core/CircularProgress';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';

import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import RefreshIcon from '@material-ui/icons/Refresh';
import FormControl from '@material-ui/core/FormControl';

import DateRangePicker from '@wojtekmaj/react-daterange-picker';

function Analysis(props){

    
    const {setDateRange,dateRange,refreshTemp,tempData,state} = props;
    return(
        <Grid item xs={12} md={12} lg={12}>
            <Grid
                container
                item
                xs={12}
                md={12}
                lg={12}
                justify="space-between"
                style={{paddingRight: 10, paddingLeft: 10, paddingTop: 10}}
            >
                <Typography>Bushfire Analysis</Typography>
                <div style={{justifyContent: 'center'}}>
                    {/* <FormControl size="small" style={{paddingRight: 10, paddingLeft: 10 }}>
                        <Select
                            labelId="station-id-label"
                            id="station-id"
                            value={props.method.stationId}
                            onChange={props.method.handleStationIdChange}
                        >
                            {props.method.validStationIds.map(id => <MenuItem value={id}>{id}</MenuItem>)}
                        </Select>
                    </FormControl> */}
                    <DateRangePicker
                        // onChange={setDateRange}
                        value={dateRange}
                    />
                     <IconButton aria-label="refresh" onClick={refreshTemp}>
                        <RefreshIcon fontSize="small" />
                    </IconButton>
                </div>
            </Grid>
            <Grid item xs={12} md={12} lg={12}>
                {(tempData.length !== 0) ? (<Chart data={{data:tempData, state:state}}/>) : (<CircularProgress/>)}
            </Grid>
        </Grid>

    )
}

export default Analysis