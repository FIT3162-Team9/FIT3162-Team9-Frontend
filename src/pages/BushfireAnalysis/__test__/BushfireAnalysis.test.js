import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Chart from "../AnalysisChart";
import Bushfire from "./..";
import Analysis from "../AnalysisContainer";

describe("Bushfire Component Test", () => {
  it("Bushfire component renders without crashing", () => {
    shallow(<Bushfire LGA={"Banyule"} station={"86068"} />);
  });

  it("Bushfirei component renders correctly", () => {
    const tree = shallow(<Bushfire LGA={"Banyule"} station={"86068"} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});

describe("Analysis Component Test", () => {
  it("Analysis component renders without crashing", () => {
    shallow(<Analysis tempData={[]} />);
  });

  it("Analysis component renders correctly", () => {
    const tree = shallow(<Analysis tempData={[]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});

describe("Analysis Component Test", () => {
  it("AnalysisChart renders without crashing", () => {
    shallow(<Chart />);
  });

  it("AnalysisChart renders correctly", () => {
    const tree = shallow(<Chart />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
