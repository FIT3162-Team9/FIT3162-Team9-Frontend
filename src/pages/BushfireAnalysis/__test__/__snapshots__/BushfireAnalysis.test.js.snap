// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analysis Component Test Analysis component renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  lg={12}
  md={12}
  xs={12}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    justify="space-between"
    lg={12}
    md={12}
    style={
      Object {
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
    xs={12}
  >
    <WithStyles(ForwardRef(Typography))>
      Bushfire Analysis
    </WithStyles(ForwardRef(Typography))>
    <div
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <DateRangePicker
        calendarIcon={
          <svg
            className="react-daterange-picker__calendar-button__icon react-daterange-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="15"
              width="15"
              x="2"
              y="2"
            />
            <line
              x1="6"
              x2="6"
              y1="0"
              y2="4"
            />
            <line
              x1="13"
              x2="13"
              y1="0"
              y2="4"
            />
          </svg>
        }
        clearIcon={
          <svg
            className="react-daterange-picker__clear-button__icon react-daterange-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="15"
              y1="4"
              y2="15"
            />
            <line
              x1="15"
              x2="4"
              y1="4"
              y2="15"
            />
          </svg>
        }
        closeCalendar={true}
        isOpen={null}
        name="daterange"
        rangeDivider="â€“"
      />
      <WithStyles(ForwardRef(IconButton))
        aria-label="refresh"
      >
        <RefreshIcon
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    lg={12}
    md={12}
    xs={12}
  >
    <WithStyles(ForwardRef(CircularProgress)) />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Analysis Component Test AnalysisChart renders correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height={300}
  width="95%"
>
  <ComposedChart
    barCategoryGap="10%"
    barGap={4}
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      dataKey="date"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="number"
      unit=""
      width={60}
      yAxisId={0}
    />
    <Tooltip
      active={false}
      allowEscapeViewBox={
        Object {
          "x": false,
          "y": false,
        }
      }
      animationDuration={400}
      animationEasing="ease"
      contentStyle={Object {}}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      filterNull={true}
      isAnimationActive={true}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      useTranslate3d={false}
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <CartesianGrid
      fill="none"
      horizontal={true}
      horizontalFill={Array []}
      horizontalPoints={Array []}
      stroke="#eee"
      strokeDasharray="5 5"
      vertical={true}
      verticalFill={Array []}
      verticalPoints={Array []}
    />
    <Area
      activeDot={false}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#addfad"
      fillOpacity="0.2"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      points={Array []}
      stackId="1"
      stroke="white"
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Area
      activeDot={false}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#87ceeb"
      fillOpacity="0.2"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      points={Array []}
      stackId="1"
      stroke="white"
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Area
      activeDot={false}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#fcf75e"
      fillOpacity="0.2"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      points={Array []}
      stackId="1"
      stroke="white"
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Area
      activeDot={false}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#ffa500"
      fillOpacity="0.2"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      points={Array []}
      stackId="1"
      stroke="white"
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Area
      activeDot={false}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#ff4040"
      fillOpacity="0.2"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      points={Array []}
      stackId="1"
      stroke="white"
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Area
      activeDot={false}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#c80815"
      fillOpacity="0.2"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      points={Array []}
      stackId="1"
      stroke="white"
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="orange"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="blue"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="green"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey=""
      dot={false}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      name="Bushfire rating"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      onMouseOver={[Function]}
      points={Array []}
      stroke="red"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
  </ComposedChart>
</ResponsiveContainer>
`;

exports[`Bushfire Component Test Bushfirei component renders correctly 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-container-6"
  maxWidth="lg"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={9}
      md={8}
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2 makeStyles-autoHeight-3"
      >
        <Analysis
          dateRange={
            Array [
              "2019-09-25T04:44:39.730Z",
              "2020-09-25T04:44:39.730Z",
            ]
          }
          refreshTemp={[Function]}
          setDateRange={[Function]}
          state={
            Object {
              "bushfireratings": true,
              "bushfirezone": true,
              "humidity": false,
              "temperature": true,
              "windspeed": false,
            }
          }
          stationId="86068"
          tempData={Array []}
        />
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={3}
      md={4}
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-sliderHeight-4"
      >
        <CustomizedSlider
          method={
            Object {
              "setDrought": [Function],
              "setHumidity": [Function],
              "setWind": [Function],
            }
          }
        />
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-filterHeight-5"
      item={true}
      lg={3}
      md={4}
      xs={12}
    >
      <BushfireFilter
        method={
          Object {
            "constLevel": Array [
              11,
              12,
              24,
              24,
              null,
              null,
            ],
            "setState": [Function],
            "state": Object {
              "bushfireratings": true,
              "bushfirezone": true,
              "humidity": false,
              "temperature": true,
              "windspeed": false,
            },
          }
        }
      />
      <About />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-filterHeight-5"
      item={true}
      lg={3}
      md={4}
      xs={12}
    >
      <Popup
        props={
          Object {
            "handleClickOpen": [Function],
            "handleClose": [Function],
            "open": false,
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
